core/templates/dev/head/pages/email_dashboard/EmailDashboard.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for oppia email dashboard page.
+ */
+
+oppia.controller('EmailDashboard', [
+  '$scope', '$http', 'EmailDashboardDataService',
+  function($scope, $http, EmailDashboardDataService) {
+    $scope.currentPageOfQueries = [];
+
+    var updateCurrentPageOfQueries = function(queries) {
+      $scope.currentPageOfQueries = queries;
+    };
+
+    var setQueryStatus = function(index, query) {
+      $scope.currentPageOfQueries[index] = query;
+    };
+
+    $scope.resetForm = function() {
+      $scope.has_not_logged_in_for_n_days = null;
+      $scope.inactive_in_last_n_days = null;
+      $scope.created_at_least_n_exps = null;
+      $scope.created_fewer_than_n_exps = null;
+      $scope.edited_at_least_n_exps = null;
+      $scope.edited_fewer_than_n_exps = null;
+    };
+
+    $scope.submitQuery = function() {
+      var data = {
+        has_not_logged_in_for_n_days: $scope.has_not_logged_in_for_n_days,
+        inactive_in_last_n_days: $scope.inactive_in_last_n_days,
+        created_at_least_n_exps: $scope.created_at_least_n_exps,
+        created_fewer_than_n_exps: $scope.created_fewer_than_n_exps,
+        edited_at_least_n_exps: $scope.edited_at_least_n_exps,
+        edited_fewer_than_n_exps: $scope.edited_fewer_than_n_exps
+      };
+      EmailDashboardDataService.submitQuery(data, updateCurrentPageOfQueries);
+      $scope.resetForm();
+      $scope.showSuccessMessage = true;
+    };
+
+    $scope.getNextPageOfQueries = function() {
+      if (EmailDashboardDataService.isNextPageAvailable()) {
+        EmailDashboardDataService.getNextQueries(updateCurrentPageOfQueries);
```

Comment:
Can you do this as a promise, instead?
```
EmailDashboardDataService.getNextQueries().then(updateCurrentPageOfQueries);
```
or
```
EmailDashboardDataService.getNextQueries().then(function(queries) {
  updateCurrentPageOfQueries(queries);
});
```

I think it reads better that way. Ditto below.